### PR TITLE
add floating point representation for numeric tooltip

### PR DESCRIPTION
**Pull request by guoci**

### CODE REVIEW

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot for your contribution, we really appreciate it :)
I have a comment about the code itself that could use a bit of explanation though

File: static/panes/compiler.ts [hunk 1]
```diff
@@ -3416,6 +3416,13 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
             result += ' = 0x' + numericValue.toString(16).toUpperCase();
         }
 
+        // Float32/64 representation.
+        new BigUint64Array(buf)[0] = BigInt(numericValue.toString());
```
[the remark this answers is not in the document]
Comment (guoci) [reply static/panes/compiler.ts:3420]: `buf` is modified.

[the remark this answers is not in the document]
Comment (RubenRBS) [reply static/panes/compiler.ts:3420]: I think a comment clarifying that would be helpful for it not to be removed in the future :)

[the remark this answers is not in the document]
Comment (guoci) [reply static/panes/compiler.ts]: added the comment

[the remark this answers is not in the document]
Comment (RubenRBS) [reply static/panes/compiler.ts]: Thanks! Im just seeing that the code is also doing the same a few lines earlier, didn't see that before, but it now makes it so much clearer :)

**Comment by partouf** [comment]
I would prefer the code to be refactored so there's no loose "new" statements, just assign it to a variable - and if that produces too many lines, put it in a new separate function

**Comment by guoci** [comment]
@partouf I have refactored the code, let me know if that is ok. I can also refactor the code earlier in that function, which was also written by me.

**Comment by guoci** [comment]
@partouf done

**Review by partouf — APPROVED** [review]

**Comment by partouf** [comment]
> @partouf done

Thanks!

**Review by RubenRBS — APPROVED** [review]
Much _much_ better now. Thanks a lot for your continued efforts :)

**Review by jeremy-rifkin — APPROVED** [review]
Looks great

**Comment by partouf** [comment]
This is now live

**Pull request merged**